### DISCUSSION TITLE
issue: Non-controlled input get focused upon error before controlled ones

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#8952] [General]
### Version Number

7.34.2

### Codesandbox/Expo snack

https://codesandbox.io/s/vigorous-fast-nqwh2z?file=/src/App.tsx

### Steps to reproduce

I expect RHF to give focus to the first element in the form that has an error but it currently gives focus to the third input.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
those issues are common due to missing ref forward.

**Comment by hornta** [comment]
I'm not sure what the resolution is. I'm using the forwardRef in the example but maybe I'm doing it in the incorrect way? I mean it focuses the field if no other non-controlled fields are erroring so the refs are working I think.

**Comment by leapful** [comment]
@hornta : Focus field on error depends on order, in your example: <input> tag will be the first DOM element attached to DOM tree while <Controller> component will be later because it will call "render" prop after first rendering.

Referrer: https://react-hook-form.com/api/useform#props

>Note: the focus order is based on the register order.

A workaround for your use case is to use <Controller> for the <input> tag too.

Working demo: https://codesandbox.io/s/practical-meadow-tth5wj?file=/src/App.tsx
